data/net/minecraft/world/level/levelgen/placement/PlacementContext.mapping

```diff
@@ -3,3 +3,12 @@ CLASS net/minecraft/world/level/levelgen/placement/PlacementContext
 		ARG 1 level
 		ARG 2 generator
 		ARG 3 topFeature
+	METHOD getBlockState (Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/block/state/BlockState;
+		ARG 1 pos
+	METHOD getCarvingMask (Lnet/minecraft/world/level/ChunkPos;Lnet/minecraft/world/level/levelgen/GenerationStep$Carving;)Lnet/minecraft/world/level/chunk/CarvingMask;
+		ARG 1 chunkPos
+		ARG 2 carving
+	METHOD getHeight (Lnet/minecraft/world/level/levelgen/Heightmap$Types;II)I
+		ARG 1 heightmap
```

Comment:
```suggestion
		ARG 1 type
```

Similarly, `Heightmap.Types` is a list of *types* used to query a heightmap, you aren't passing in the heightmap itself.